typescript/sdk/src/router/RouterApps.ts

```diff
@@ -44,11 +44,17 @@ export abstract class RouterApp<
     return this.foreignDeployments[chainName];
   }
 
-  override remoteChains(chainName: string): string[] {
-    return [
-      ...super.remoteChains(chainName),
-      ...Object.keys(this.foreignDeployments),
-    ].filter((chain) => chain !== chainName);
+  // check onchain for remote enrollments
+  override async remoteChains(chainName: string): Promise<ChainName[]> {
+    const router = this.router(this.contractsMap[chainName]);
+    const domains = (await router.domains()).map((domain) => {
```

Comment:
Nit: rename this, the name made me thing it was a domainId

Reply:
done